### ISSUE TITLE
[SSL: WRONG_VERSION_NUMBER] errors with Indian video site

### ISSUE THREAD

**Issue by velw** [yt-dlp/yt-dlp#4504]
### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.07.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

When downloading videos from a specific site hosted in India, I've started getting "[SSL: WRONG_VERSION_NUMBER]" errors. 

The website is region restricted so can only be access from an IP address within India (a different error shows up very quickly if you're not on an Indian IP address).

There were no issues using yt-dlp.exe with this site until mid-May. I've been busy and just came back to try and download again today, and now am hitting this error. I've also had to switch VPN providers due to the previous provider halting India access, but the video and URL load fine within the browser, albeit a little slow.

I'm getting the errors right now even when just trying to get a list of available formats. It's weird. Occasionally a different error shows up, and once I did get the list of formats to display. I also got a lower-resolution version of the video to start downloading once as well. That makes me wonder if this is really an SSL version error or something else. The downloads (or download attempts) are also painfully slow, but I'm putting that down to the VPN.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['--no-check-certificate', '-vU', '-F', '--socket-timeout', '300', '--write-sub', 'https://www.hotstar.com/in/tv/mahabharat/435/shakuni-stabs-abhimanyu/1000012005']
[debug] Encodings: locale cp932, fs utf-8, pref cp932, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.07.18 [135f05e] (win32_exe)
[debug] Python 3.8.10 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: avprobe -bsfs
[debug] exe versions: ffmpeg git-2019-12-26-b0d0d7e
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.07.18, Current version: 2022.07.18
yt-dlp is up to date (2022.07.18)
[debug] [hotstar] Extracting URL: https://www.hotstar.com/in/tv/mahabharat/435/shakuni-stabs-abhimanyu/1000012005
[hotstar] 1000012005: Downloading JSON metadata
[hotstar] 1000012005: Downloading webpage
[hotstar] 1000012005: Downloading token
[hotstar] 1000012005: Downloading JSON metadata
ERROR: [hotstar] 1000012005: Unable to download JSON metadata: <urlopen error [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1131)> (caused by URLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1131)')))
  File "yt_dlp\extractor\common.py", line 644, in extract
  File "yt_dlp\extractor\hotstar.py", line 176, in _real_extract
  File "yt_dlp\extractor\hotstar.py", line 74, in _call_api_v2
  File "yt_dlp\extractor\hotstar.py", line 48, in _call_api_impl
  File "yt_dlp\extractor\common.py", line 1004, in download_content
  File "yt_dlp\extractor\common.py", line 968, in download_handle
  File "yt_dlp\extractor\common.py", line 838, in _download_webpage_handle
  File "yt_dlp\extractor\common.py", line 795, in _request_webpage

  File "urllib\request.py", line 1354, in do_open
  File "http\client.py", line 1252, in request
  File "http\client.py", line 1298, in _send_request
  File "http\client.py", line 1247, in endheaders
  File "http\client.py", line 1007, in _send_output
  File "http\client.py", line 947, in send
  File "http\client.py", line 1421, in connect
  File "ssl.py", line 500, in wrap_socket
  File "ssl.py", line 1040, in _create
  File "ssl.py", line 1309, in do_handshake
ssl.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1131)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "yt_dlp\extractor\common.py", line 777, in _request_webpage
  File "yt_dlp\YoutubeDL.py", line 3639, in urlopen
  File "urllib\request.py", line 525, in open
  File "urllib\request.py", line 542, in _open
  File "urllib\request.py", line 502, in _call_chain
  File "yt_dlp\utils.py", line 1455, in https_open
  File "urllib\request.py", line 1357, in do_open
urllib.error.URLError: <urlopen error [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1131)>
```

**Comment by gamer191** [comment]
That shouldn't be happening, given you passed `--no-check-certificate`. How strange

**Comment by coletdjnz** [comment]
iirc this error happens when you try to use SSL/HTTPS on HTTP site. 

Maybe something is intercepting your connection to point you to some HTTP webpage instead of the real site?

**Comment by velw** [comment]
> Maybe something is intercepting your connection to point you to some HTTP site instead of the real site?
> That or maybe there is an issue with the url or server being used for the json metadata download.

These are possible. I didn't have this issue before. It's possible that in the last couple of months the site has changed its configuration (although not so likely tbh, it doesn't look like they've revamped it at all). The VPN is more likely to be the issue, and I did manage to get one file to download in the end last night -- although only by writing a batch file wrapper for yt-dlp.exe that retries the download persistently whenever it errors out until it finally succeeds. Once the file had actually started downloading, that download failed every 10-15 minutes or so but resumed after a couple of retries, without yt-dlp.exe exiting. I'm having real trouble downloading the subtitle file to go with it, too - I could only get the video to download by giving up on the subtitle file, and when I try to download that separately it keeps failing with HTTP 403... I had no issues downloading the subtitle files at all a couple of months ago.

Given that one download did work in the end (with many, many retries), is there actually an SSL problem at all? If the message is a red herring it really makes troubleshooting downloads harder than ideal. If this was an issue with server configuration you'd think it would never succeed. It seems more like this is some kind of timeout being (mis)reported as an SSL problem.

I'm trying to get set up with a different VPN in the hope that this is an intermittent VPN capacity problem and that a different one will work better. In the meantime, though, it seems odd that yt-dlp.exe is throwing the SSL error when that doesn't seem to be what the problem is, and also when `--no-check-certificate` is specified. Or will it still run into SSL issues because (whether or not it checks the certificate) it still has to negotiate an HTTPS connection? (I did try sending the url just as HTTP instead of HTTPS but it didn't make any difference.)

**Comment by pukkandan** [comment]
> I'm trying to get set up with a different VPN

If you were using a VPN in the original report, it is most likely the VPN's fault

```
❯ yt-dlp -s https://www.hotstar.com/in/tv/mahabharat/435/shakuni-stabs-abhimanyu/1000012005
[hotstar] 1000012005: Downloading JSON metadata
[hotstar] 1000012005: Downloading webpage
[hotstar] 1000012005: Downloading token
[hotstar] 1000012005: Downloading JSON metadata
[hotstar] 1000012005: Downloading m3u8 information
[hotstar] 1000012005: Downloading MPD manifest
[hotstar] 1000012005: Downloading MPD manifest
[hotstar] 1000012005: Downloading MPD manifest
[hotstar] 1000012005: Downloading m3u8 information
[hotstar] 1000012005: Downloading MPD manifest
[hotstar] 1000012005: Downloading m3u8 information
[hotstar] 1000012005: Downloading m3u8 information
[hotstar] 1000012005: Downloading MPD manifest
[hotstar] 1000012005: Downloading m3u8 information
[hotstar] 1000012005: Downloading m3u8 information
[hotstar] 1000012005: Downloading m3u8 information
[info] 1000012005: Downloading 1 format(s): sdr-dash-video-avc1-7+sdr-dash-audio-und-mp4a-2-1
```

**Comment by velw** [comment]
Why did you add the "incomplete" tag to this report? I stated from the start that I was using a VPN (and why). It's also not clear why yt-dlp.exe is reporting that it has failed due to an SSL problem when that isn't the case. 

Just closing this or marking it as "not a problem" or falsely marking that the person who reported it didn't do so correctly is unhelpful.

**Comment by pukkandan** [comment]
The VPN is creating the SSL issue. The incomplete label can be removed if you can reproduce the issue without any VPN

**Comment by velw** [comment]
> The incomplete label can be removed if you can reproduce the issue without any VPN

If I use a different VPN and the problem goes away, the problem was the VPN. If I use a different VPN and the problem still exists, the problem isn't the VPN. And if the VPN was "creating the SSL issue", it wouldn't be intermittent. It would fail every time.

**Comment by pukkandan** [comment]
ok, does it go away?

**Comment by velw** [comment]
> ok, does it go away?

I don't know yet. I've been battling with another VPN provider since yesterday over payment issues to get an account set up. But I will report here either way.

**Comment by pukkandan** [comment]
>  Or will it still run into SSL issues because (whether or not it checks the certificate) it still has to negotiate an HTTPS connection? (I did try sending the url just as HTTP instead of HTTPS but it didn't make any difference.)

It can, depending on the server and VPN configuration

**Comment by pukkandan** [comment]
> I don't know yet. I've been battling with another VPN provider since yesterday over payment issues to get an account set up. But I will report here either way.

So until you report back, the issue is incomplete. That doesn't mean you didn't open the issue correctly, just that more info is needed from your side

**Comment by velw** [comment]
> That doesn't mean you didn't open the issue correctly, just that more info is needed from your side

Ok, fair enough. In the meantime, if it is an SSL issue caused by the VPN, why wouldn't it fail with that error every time? A genuine SSL issue wouldn't be intermittent, surely. A timeout issue, yes... but in that case shouldn't the error be reported as a timeout?

**Comment by pukkandan** [comment]
The error message is being produced by OpenSSL. I can't say for sure that it is the correct message, but   the error is definitely happening during the SSL handshake.

As for the intermittent nature of the error; badly configured VPNs/proxies can give all kinds of weird and intermittent errors. I can't possibly try and diagnose what the issue in the network actually is

On a sidenote, can you try `--legacy-server-connect`? The error message doesn't match with the errors it is supposed to handle, but might as well try just in case

**Comment by velw** [comment]
Thank you for keeping this open. 

I've now been able to try this out with a different VPN service and you're right, it was the VPN. The new service works very smoothly and downloads the whole file in about five minutes, without any errors. _I will be cancelling the other one!_

> On a sidenote, can you try `--legacy-server-connect`? The error message doesn't match with the errors it is supposed to handle, but might as well try just in case

Out of curiosity I did try this even after seeing that the other VPN worked, to see if using this option would have solved the problem, but it didn't. I tried  `--force-ipv4`  too, as I'd seen that suggested elsewhere, but it also didn't help.

This does seem to have been nothing to do with yt-dlp, though, and everything to do with the VPN. Apologies for the false alarm, and I hope some of the details noted here might help anyone else who runs across the same kind of issue in the future figure out where the problem really lies. Thank you for the information and assistance!
